### DISCUSSION TITLE
mount a component in the action directly?

### DISCUSSION THREAD

**Discussion by zw963** [luckyframework/lucky#1887] [Q&A] [answered]
I have a action/page pair like this:

```crystal
# some action
class Htmx::Foo < BrowserAction

# ...
html(
            Htmx::FooPage,
            id: university.id.to_s,
            column_name: param_value[0],
            column_value: param_value[1],
            action: "/htmx/foo"
          )
end
# ...
```

```crystal
class Htmx::FooPage < NoLayout
  needs id : String
  needs column_name : String
  needs column_value : String
  needs action : String

  def content
    mount(
      ClickEditTD,
      id: id,
      column_value: column_value,
      column_name: column_name,
      action: action,
      tooltip: show_ranking_number(column_value)
    )
  end
end
```

As you can see, this action used with htmx, the only purpose is, pass all args into view, 
and mount the component in the view for render the partial.

Can we mount the component in the action diretly instead?

For this use cases (work with htmx), render use `html` is really useless, even, 

i have to create a new `NoLayout`(empty layout) for only this purpose, thanks!

**Comment by jwoertink** [comment, marked as answer]
You can call `component` from `BrowserAction`. https://luckyframework.org/guides/http-and-routing/request-and-response#handling-responses

```crystal
class Htmx::Foo < BrowserAction

# ...
component(
            ClickEditTD,
            id: university.id.to_s,
            column_name: param_value[0],
            column_value: param_value[1],
            action: "/htmx/foo"
          )
end
```

**Reply by zw963** [reply]
Hi, is there any reason for use different name for same functionality(`mount` in page, but `component` in action)?

**Reply by jwoertink** [reply]
Yes, because they work differently. `component` takes a different set of arguments to return HTTP status, and content_type as where `mount` doesn't. I didn't want to confuse the two.

**Reply by zw963** [reply]
In fact, i consider if both use `component` as name, don't confuse anyway (at least for me), one used in view, and another use in action.
